### PR TITLE
Update API paramters for Untagging Golden Image, fixed wired device p…

### PR DESCRIPTION
**Pull request by Abhishek-121**
…rovision issue just after device addition.

In this commit - 

-- Resolve the issue of Wired Device Provisioning just after device addition, it will check till device state and collectionStatus become managed and hostname appears.

-- Fix the issue of Untagging Golden image in SWIM module by changing the API parameters.

-- Add more log messages for better understanding of code.

-- Give support of Site name while distributing image to the devices associated with site.

### CODE REVIEW

File: plugins/modules/inventory_intent.py [hunk 1]
```diff
@@ -1472,7 +1472,7 @@ def provisioned_wired_device(self):
                 while True:
                     response = self.get_device_response(device_ip)
                     self.log("Device is in {0} state waiting for Managed State.".format(response['managementState']))
-                    if response['managementState'] == "Managed":
+                    if response['managementState'] == "Managed" and response['collectionStatus'] == "Managed" and response["hostname"]:
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:1481]: can we split into 2 or 3 lines?

```
if (
    response.get('managementState') == "Managed" 
    and response.get('collectionStatus') == "Managed" 
    and response.get("hostname")
):
    break
```

File: plugins/modules/swim_intent.py [hunk 2]
```diff
@@ -689,6 +688,14 @@ def get_have(self):
         if self.want.get("distribution_details"):
             have = {}
             distribution_details = self.want.get("distribution_details")
+            site_name = distribution_details.get("site_name")
+            if site_name:
+                site_exists = False
+                (site_exists, site_id) = self.site_exists(site_name)
+                if site_exists:
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:695]: Add a new line before if block

File: plugins/modules/inventory_intent.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:1481]: added the above code.

File: plugins/modules/swim_intent.py (hunk 2 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:695]: added.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**